### PR TITLE
fallback to get hub host from hub kubeconfig secret when detaching cluster

### PR DESCRIPTION
**Pull request by zhujian7**

### CODE REVIEW

File: pkg/operators/klusterlet/controllers/klusterletcontroller/klusterlet_managed_reconcile.go [hunk 1]
```diff
@@ -178,17 +178,16 @@ func (r *managedReconcile) cleanUpAppliedManifestWorks(ctx context.Context, klus
 		return nil
 	}
 
-	bootstrapKubeConfigSecret, err := r.kubeClient.CoreV1().Secrets(config.AgentNamespace).Get(ctx, config.BootStrapKubeConfigSecret, metav1.GetOptions{})
+	hubHost, err := r.getHostFromKubeconfigSecret(ctx, config.AgentNamespace, config.BootStrapKubeConfigSecret)
```
Comment (qiujian16) [inline pkg/operators/klusterlet/controllers/klusterletcontroller/klusterlet_managed_reconcile.go:181]: this is tricky...what happens when all secret are missing? Actually should we use agentID only to handle applied manifestwork regardless of hubhash, what is the side effect?

Comment (zhujian7) [reply pkg/operators/klusterlet/controllers/klusterletcontroller/klusterlet_managed_reconcile.go:181]: I think we just introduced the `appliedmanifestwork.spec.agentID` last release, I'm not sure all appliedmanifestwork, upgrading case, will have the value

Comment (qiujian16) [reply pkg/operators/klusterlet/controllers/klusterletcontroller/klusterlet_managed_reconcile.go:181]: work agent will update agent id for each appliedmanifestwork

**Comment by zhujian7** [comment]
https://github.com/open-cluster-management-io/registration-operator/pull/321 will fix this.
/close

**Pull request closed without merging**